### PR TITLE
SOLR-17197: Fix getting fieldType by its name in FileBasedSpellChecker

### PR DESCRIPTION
**Pull request by AndreyBozhko**
https://issues.apache.org/jira/browse/SOLR-17197

# Description

Fix logic in FileBasedSpellChecker that uses wrong method to retrieve fieldType by its name

# Tests

There are a few existing tests in FileBasedSpellCheckerTest - but those tests refer to `schema.xml` which defines the field `teststop` and fieldType by the same name `teststop` (so the bug stays hidden).

I changed the tests to refer to `schema-spellchecker.xml` instead (to avoid changes to other unrelated tests that may depend on `schema.xml`), and made sure that the `teststop` field and its type do not have exactly the same name.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/CHANGES.txt [hunk 1]
```diff
@@ -27,7 +27,8 @@ Optimizations
 
 Bug Fixes
 ---------------------
-(No changes)
+
+* PR#2329: Fix getting fieldType by its name in FileBasedSpellChecker (Andrey Bozhko)
```
Comment (epugh) [inline solr/CHANGES.txt:31]: I *believe* we normally create a JIRA for these user facing, so we can track which version of Solr it gets released.   The pattern has recently become a bit muddier...  non user facing changes sometimes skipping JIRA..   If you add a JIRA, then that makes it easier for me to merge this!

File: solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml [hunk 2]
```diff
@@ -49,9 +49,22 @@
     </analyzer>
   </fieldType>
 
+  <fieldType name="teststop_type" class="solr.TextField">
```
Comment (epugh) [inline solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml:52]: I haven't checked in more detail, but don't we normally camelCase?   testStopType ???

**Review by epugh — CHANGES_REQUESTED** [review]
Looks like a nice fix!   Do we have this same issue in lots of other places do you know?

File: solr/CHANGES.txt (hunk 1 above)
Comment (epugh) [reply solr/CHANGES.txt:31]: i confirmed that yes, please open a jira...

Comment (AndreyBozhko) [reply solr/CHANGES.txt:31]: I submitted a request for a JIRA account, will open the issue once the account is set up

File: solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml (hunk 2 above)
Comment (AndreyBozhko) [reply solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml:52]: I saw that in different test files, the fields and field types appear in either camel case or snake case (an sometimes even in kebab case).

I personally use snake case, and I believe the Ref Guide examples also suggest snake case (e.g., field types `text_general` or `text_ws`).

**Review by AndreyBozhko — COMMENTED** [review]
> Do we have this same issue in lots of other places do you know?

I searched for the usage of methods like `getFieldTypeByName`, `getFieldTypeNoEx`, etc. - and at a quick glance, nothing else stood out that was obviously wrong.

File: solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml (hunk 2 above)
Comment (epugh) [reply solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml:52]: thanks for taking the time to check the Ref Guide!

Comment (epugh) [reply solr/core/src/test-files/solr/collection1/conf/schema-spellchecker.xml:52]: we probably should `snake_case` it everywhere......   in all of our test-files..  to have a single pattern.

**Pull request merged**